server/paymentProviders/utils.js

```diff
@@ -35,125 +37,157 @@ export const persistTransaction = async (
   transactionToken,
   providerTransaction,
   isRefund = false,
+  fromAuthorizationId = null,
 ) => {
   if (amount === 0) {
     return;
   }
 
+  const data = { ...omit(providerTransaction, ['id']), id: transactionToken };
   const UserId = virtualCard.UserId;
   const host = virtualCard.host;
   const collective = virtualCard.collective;
+  const vendor = await getOrCreateVendor(vendorProviderId, vendorName);
+  const hostCurrencyFxRate = await getFxRate('USD', host.currency);
+  const description = `Virtual Card charge: ${vendor.name}`;
+
+  if (fromAuthorizationId) {
+    const processingExpense = await models.Expense.findOne({
+      where: {
+        VirtualCardId: virtualCard.id,
+        data: { authorizationId: fromAuthorizationId },
+      },
+    });
+
+    if (processingExpense && processingExpense.status === ExpenseStatus.PROCESSING) {
+      await processingExpense.update({
+        status: ExpenseStatus.PAID,
+        data,
+      });
```

Comment:
By not using the `setPaid` utility, we skip setting some things like setting `HostCollectiveId: collective.HostCollectiveId`